mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java

```diff
@@ -1826,4 +1829,49 @@ public Source resolve(String href, String base) throws TransformerException {
         }
     }
 
+    /**
+     * Resolves arbitrary static methods of arbitrary classes. Parameters are considerd to be of type
+     * {@link java.lang.String}.
+     * <br/><br/>
+     * <strong>Invocation</strong>
+     * <pre><code>function:&lt;class name&gt;:&lt;method name&gt;:&lt;param1&gt;:&lt;param2&gt;</code></pre>
+     * <br/>
+     * <strong>Example</strong>
+     * <pre><code>function:de.uni_jena.thunibib.user.ThUniBibUtils:getLeadId:id_connection:foobar;</code></pre>
+     *
+     * @author shermann (Silvio Hermann)
+     * */
+    private static class MCRFunctionResolver implements URIResolver {
+        @Override
+        public Source resolve(String href, String base) throws TransformerException {
+            LOGGER.debug("Resolving {}", href);
+
+            String[] parts = href.split(":");
+            String className = parts[1];
+            String methodName = parts[2];
+
+            int numParams = parts.length - 3;
```

Comment:
Could all be simplified by:

`Arrays.stream(parts).skip(2).toArray(String[]::new)`